source/extensions/filters/common/lua/lua.h

```diff
@@ -69,6 +69,24 @@ namespace Lua {
   lua_pushnumber(state, val);                                                                      \
   lua_settable(state, -3);
 
+/**
+ * Get absl::string_view from Lua string. This checks if the argument at index is a string
+ * and build an absl::string_view from it.
+ * @param state the current Lua state.
+ * @param index the index of argument.
+ * @return absl::string_view of Lua string with proper string length.
+ **/
+inline absl::string_view getStringViewFromLuaString(lua_State* state, int index) {
+  size_t input_size = 0;
+  // When the argument at index in Lua state is not a string, for example, giving a table to
+  // logTrace (which uses this function under the hood), Lua script exits with an error like the
+  // following: "[string \"...\"]:3: bad argument #1 to 'logTrace' (string expected, got table)".
+  // However,`luaL_checklstring` accepts a number as its argument and implicitly converts it to a
+  // string, since Lua provides automatic conversion between string and number values at run time
```

Comment:
```suggestion
  // string, since Lua provides automatic conversion between string and number values at run time
  // (https://www.lua.org/manual/5.1/manual.html#2.2.1).
```